app/models/identity.rb

```diff
@@ -23,68 +24,38 @@ def authenticate unencrypted_password
     end
   end
 
-  # Generates a reset code (and saves the identity without validation)
-  def generate_reset_code!(expiration_period=DEFAULT_RESET_CODE_EXPIRATION_PERIOD)
-    self.reset_code = SecureRandom.hex(16)
-    if !expiration_period.nil?
-      self.reset_code_expires_at = DateTime.now + expiration_period
-    else
-      self.reset_code_expires_at = nil
-    end
-    self.save!(validate: false)
-    self.reset_code
+  # Returns true if the user is due for resetting their password
+  def password_expired?
+    !password_expires_at.nil? && password_expires_at <= DateTime.now
   end
 
-  # Returns true iff the code is valid and matches
-  def reset_code_valid?(code)
-    return true if self.reset_code == code && \
-                     (self.reset_code_expires_at.nil? || \
-                     self.reset_code_expires_at >= DateTime.now)
-    false
-  end
-
-  # Sets the password (and saves the identity)
-  def set_password!(password, password_confirmation)
-    # Always invalidate reset codes when the password changes
-    self.reset_code = nil
-    self.reset_code_expires_at = nil
-
+  # Sets the password and the password expiration date; Returns nil
+  def set_password(password,
```

Comment:
With the destroying of `reset_code` and this method's sole use from a handler, `set_password` is kind of looking like it should be a `SetPassword` routine?